### PR TITLE
Swallow and log any BaseException in ImmediateBackend other than KeyboardInterrupt

### LINKED ISSUE #50: Allow `ImmediateBackend` to throw its errors
If a task fails using the `ImmediateBackend`, this exception is currently swallowed and isn't accessible anywhere. This should be rectified by adding a `logger.exception` call to ensure the error is seen, and captured by error monitoring. This matches the behaviour of the database backend.

### PR DESCRIPTION
**Pull request by jhthompson**
Fixes #50

### CODE REVIEW

File: django_tasks/backends/immediate.py [hunk 1]
```diff
@@ -37,14 +37,21 @@ def enqueue(
...
-        except Exception as e:
+        except BaseException as e:
             try:
                 result = exception_to_dict(e)
             except Exception:
                 logger.exception("Task id=%s unable to save exception", result_id)
                 result = None
+
+            # Use `.exception` to integrate with error monitoring tools (eg Sentry)
+            logger.exception("Task execution failed: %s", e)
```
Comment (jhthompson) [inline django_tasks/backends/immediate.py:48]: Not sure if this message could be improved or not. I was looking at [these lines from db_worker](https://github.com/RealOrangeOne/django-tasks/blob/master/django_tasks/backends/database/management/commands/db_worker.py#L126-L132) as reference:

```python
# Use `.exception` to integrate with error monitoring tools (eg Sentry)
logger.exception(
    "Task id=%s path=%s state=%s",
    db_task_result.id,
    db_task_result.task_path,
    ResultStatus.FAILED,
)
```

File: django_tasks/backends/immediate.py [hunk 2]
```diff
@@ -37,14 +37,21 @@ def enqueue(
...
             status = ResultStatus.FAILED
 
+            # If the user tried to terminate, let them
+            if isinstance(e, KeyboardInterrupt):
+                raise
```
Comment (jhthompson) [inline django_tasks/backends/immediate.py:53]: Similarly [inspired from db_worker](https://github.com/RealOrangeOne/django-tasks/blob/master/django_tasks/backends/database/management/commands/db_worker.py#L135-L137)

File: django_tasks/backends/immediate.py (hunk 1 above)
Comment (RealOrangeOne) [reply django_tasks/backends/immediate.py:48]: `task.module_path` should do it. I agree it'd be nice to use `logfmt` formatting if we can.

File: django_tasks/backends/immediate.py (hunk 2 above)
Comment (RealOrangeOne) [reply django_tasks/backends/immediate.py:53]: I think this is fine. As part of #59 I'd like to add tests to make sure this all works correctly as we'd expect, but that can come later.

File: django_tasks/backends/immediate.py (hunk 1 above)
Comment (RealOrangeOne) [reply django_tasks/backends/immediate.py:48]: There's a module path on the `Task` you can use.

`logfmt` would also be great - there should be everything available here to replicate the message, you just might need to move the `TaskResult` creation above the execution for convenience.

Comment (jhthompson) [reply django_tasks/backends/immediate.py]: Pushed an improvement, I just used the `result_id` for the ID, is that fine?

**Pull request merged**